src/python_testing/TC_MCORE_FS_1_3.py

```diff
@@ -0,0 +1,158 @@
+                 TestStep(2, "TH_FSA commissions TH_SED_TH onto TH_FSAs fabric and generates a UniqueID."),
+                 TestStep(3, "Follow manufacturer provided instructions to enable DUT_FSA to synchronize TH_SED_TH onto DUT_FSAs fabric."),
+                 TestStep(4, "DUT_FSA synchronizes TH_SED_TH onto DUT_FSAs fabric and copies the UniqueID presented by TH_FSAs Bridged Device Basic Information Cluster.")]
+        return steps
+
+    @async_test_body
+    async def test_TC_MCORE_FS_1_3(self):
+        self.is_ci = self.check_pics('PICS_SDK_CI_ONLY')
+        self.print_step(0, "Commissioning DUT to TH, already done")
+        self.step(1)
+        # These steps are not explicitly in step 1, but they help identify the dynamically added endpoint in step 1.
+        root_node_endpoint = 0
+        root_part_list = await self.read_single_attribute_check_success(cluster=Clusters.Descriptor, attribute=Clusters.Descriptor.Attributes.PartsList, endpoint=root_node_endpoint)
+        set_of_endpoints_before_adding_device = set(root_part_list)
+
+        discriminator, setup_pin_code = await self.create_device_for_dut_ecosystem()
+        self.wait_for_user_input(
+            prompt_msg=f"Using the DUT vendor's provided interface, commission the device using the following parameters:\n"
+                        f"- discriminator: {discriminator}\n"
+                        f"- setupPinCode: {setup_pin_code}\n"
+                        f"If using FabricSync Admin, you may type:\n"
+                        f">>> pairing onnetwork 111 {setup_pin_code}")
+        
+        root_part_list = await self.read_single_attribute_check_success(cluster=Clusters.Descriptor, attribute=Clusters.Descriptor.Attributes.PartsList, endpoint=root_node_endpoint)
+        set_of_endpoints_after_adding_device = set(root_part_list)
+                                                  
+        asserts.assert_true(set_of_endpoints_after_adding_device.issuperset(set_of_endpoints_before_adding_device), "Expected only new endpoints to be added")
+        unique_endpoints_set = set_of_endpoints_after_adding_device - set_of_endpoints_before_adding_device
+        asserts.assert_equal(len(unique_endpoints_set), 1, "Expected only one new endpoint")
+        newly_added_endpoint = list(unique_endpoints_set)[0]
+
+        th_sed_dut_unique_id = await self.read_single_attribute_check_success(cluster=Clusters.BridgedDeviceBasicInformation, attribute=Clusters.BridgedDeviceBasicInformation.Attributes.UniqueID, endpoint=newly_added_endpoint)
+        asserts.assert_true(type_matches(th_sed_dut_unique_id, str), "UniqueID should be a string")
+        asserts.assert_true(th_sed_dut_unique_id, "UniqueID should not be an empty string")
+
+        self.step(2)
+        await self.create_and_commission_device_for_th_ecosystem()
+        # TODO During setup we need to create the TH_FSA device where we would commission device created
```

Comment:
Is this just a separate device that's meant to commission the DUT? Why can this not just be a separate controller on a different fabric?

Reply:
There is something missing from test plan that I will add.

Where it sat `TH_FSA`, that needs to be two device `TH` and `TH_FSA`. The reason we cannot use a different fabric here is that when the `DUT_FSA` talks to the `TH_FSA` it needs to act like a fabric sync device which chip-repl is not capable of doing at the moment

Reply:
OK, so there needs to be an app where you can trigger the adding of a new BridgedDeviceBasicInformation endpoint?

Can you open a followup for this?

Reply:
https://github.com/CHIP-Specifications/chip-test-plans/issues/4375